x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -256,22 +256,41 @@ export class AlertsClient {
   }
 
   public async update({ id, data }: UpdateOptions): Promise<PartialAlert> {
-    const decryptedAlertSavedObject = await this.encryptedSavedObjectsPlugin.getDecryptedAsInternalUser<
-      RawAlert
-    >('alert', id, { namespace: this.namespace });
-    const updateResult = await this.updateAlert({ id, data }, decryptedAlertSavedObject);
-
-    if (
-      updateResult.scheduledTaskId &&
-      !isEqual(decryptedAlertSavedObject.attributes.schedule, updateResult.schedule)
-    ) {
-      this.taskManager.runNow(updateResult.scheduledTaskId).catch((err: Error) => {
-        this.logger.error(
-          `Alert update failed to run its underlying task. TaskManager runNow failed with Error: ${err.message}`
-        );
-      });
+    let alertSavedObject: SavedObject<RawAlert>;
+
+    try {
+      alertSavedObject = await this.encryptedSavedObjectsPlugin.getDecryptedAsInternalUser<
+        RawAlert
+      >('alert', id, { namespace: this.namespace });
+    } catch (e) {
+      // We'll skip invalidating the API key since we failed to load the decrypted saved object
+      this.logger.error(
+        `update(): Failed to load API key to invalidate on alert ${id}: ${e.message}`
+      );
+      // Still attempt to load the object using SOC
+      alertSavedObject = await this.savedObjectsClient.get<RawAlert>('alert', id);
     }
 
+    const updateResult = await this.updateAlert({ id, data }, alertSavedObject);
+
+    await Promise.all([
```

Comment:
I think technically, we don't really need a `Promise.all()` here.  The old key should be invalidated, but no one should really be dependent on when that happens.  Especially when it's paired with the TM runNow().  But this is really a nit.  Worthy of perhaps a TODO comment to separate these - will require a bit of work to do error-checking on the invalidate key, and what to do when it fails?  Not worth fixing ATM, I don't think.

Reply:
I see your point, there's good and bad parts on both sides of doing an `await` here. My original thinking was that it facilitates tests knowing when the request is finished, all the underlying tasks are finished as well. This also facilitates controlling the flow in bulk if ever doing batches to avoid errors possibly caused by underlying tasks (ex: SIEM bulking requests for ~300 alerts).

On the other hand, it doesn't make sense slowing down the user and we have to add checks to make sure the error doesn't bubble up.

I'm happy to create an issue for this if we want to revisit this later?

Reply:
Let's wait to see if it causes some kind of perf issue.  We should see these show up in APM traces, we'll get a sense for if there is something we'll need to change.